web/packages/teleterm/src/ui/AccessRequestCheckout/useAccessRequestCheckout.ts

```diff
@@ -145,9 +154,21 @@ export default function useAccessRequestCheckout() {
     }
   }, [showCheckout, hasExited, createRequestAttempt.status]);
 
-  function getPendingAccessRequestsPerResource(
-    pendingRequest: PendingAccessRequest
-  ): PendingListItemWithOriginalItem[] {
+  /**
+   *
+   * @param pendingRequest holds a list or map of resources to process
+   * @param excludeSubResourceParentResource when true, resources that have
+   * subresources, will be excluded from the returned list. eg:
+   * if a kube_cluster resource has a list of namespaces (subresources),
+   * then if this flag is true, kube_cluster will be excluded from the result.
+   */
+  function getPendingAccessRequestsPerResource({
+    pendingRequest,
+    excludeSubResourceParentResource = false,
```

Comment:
I feel like this should be rethought a little bit in order to find a simpler way of representing this concept.

`getPendingAccessRequestsPerResource` is called three times within this hook. It's always called with `pendingRequest`. Two times `excludeSubResourceParentResource` is set to `true` and one time it's set to `false` when it's called to return `data` from the hook.

Instead, I think we could experiment with the following:

* Get rid of `excludeSubResourceParentResource`, call `getPendingAccessRequestsPerResource` and assign the result to a variable, I guess `pendingAccessRequestsPerResource`? 
* Filter through `pendingAccessRequestsPerResource` once to remove resources that have subresources and assign the result to a var, say `pendingAccessRequestsPerResourceWithoutParentResources`.
* Now we remove some special behavior from `getPendingAccessRequestsPerResource` that is controlled merely through a bool arg and instead we can pass specific variables to where they're needed.

Reply:
i got so confused with naming, that this is what i decided: https://github.com/gravitational/teleport/pull/47347/commits/2b24750a3fed7337b59366ce973cd22d8342f5cb

Reply:
This does help with understanding what happens within `useAccessRequestCheckout`.